### PR TITLE
Update api to allow application service timestamp massaging

### PR DESCRIPTION
**Pull request by non-Jedi**
All state events now routed through send_state_event to make change to include timestamp kwarg easier. Added kwarg for timestamp massaging to all api methods sending events to the server.

Addresses #36.

### CODE REVIEW

**Comment by non-Jedi** [comment]
This PR is ready now.

**Comment by Half-Shot** [comment]
ok to test

File: tox.ini [hunk 1]
```diff
@@ -1,5 +1,5 @@
 [tox]
-envlist = py27,py34,py35,packaging,pep8
+envlist = py27,py34,py35,py36,packaging,pep8
```
Comment (Half-Shot) [inline tox.ini]: Could you leave this off for now, we might look at it in the future for a PR but right now I don't want to start supporting 3.6 without a proper review.

Comment (non-Jedi) [inline tox.ini]: Reverted. But I can squash all commits if you'd prefer.

**Comment by Half-Shot** [comment]
One last feature request that only just occured to me, is it possible to check to see that we are an AS client to validate if setting a timestamp makes sense? I worry it might fail silently otherwise.

**Comment by non-Jedi** [comment]
Only way I can think of off the top of my head is checking `self.identity`. But that only works if there are no use cases for timestamp massaging messages sent from the bridge itself. I can't think of any, and I don't know of any in existing applications. Unfortunately I don't think there's any way of testing whether a particular token belongs to a user or as outside of trying an operation only an as would be permitted (identity assertion or timestamp massaging).

The other option would be including an application_service bool that gets passed to the api when it's created...

**Comment by Half-Shot** [comment]
Fair enough, I'll just have to hope that synapse will freak out if the timestamp is unexpected.

**Comment by Half-Shot** [comment]
>The one other thing we could do is split the AS api off into a separate class like in #108. If we have that level of separation, it would be much more difficult for a user to make the mistake you're describing.

Nah, that's a waste really. Only a nice to have.

**Comment by Half-Shot** [comment]
LGTM. Merging,

**Pull request merged**